File: actin-common/src/test/java/com/hartwig/actin/clinical/feed/ClinicalFeedValidationTest.java
package com.hartwig.actin.clinical.feed;

import java.util.List;

import com.google.common.collect.Lists;
import com.hartwig.actin.clinical.feed.patient.ImmutablePatientEntry;
import com.hartwig.actin.clinical.feed.patient.PatientEntry;

import org.junit.Test;

public class ClinicalFeedValidationTest {

    @Test
    public void standardTestFeedIsValid() {
        ClinicalFeedValidation.validate(TestFeedFactory.createTestClinicalFeed());
    }

    @Test(expected = IllegalStateException.class)
    public void doNotAllowDuplicatePatients() {
        ClinicalFeed basisFeed = TestFeedFactory.createTestClinicalFeed();

        List<PatientEntry> patients = Lists.newArrayList(ImmutablePatientEntry.builder().from(basisFeed.patientEntries().get(0)).build(),
                ImmutablePatientEntry.builder().from(basisFeed.patientEntries().get(0)).build());

        ClinicalFeedValidation.validate(ImmutableClinicalFeed.builder().from(basisFeed).patientEntries(patients).build());
    }
}

File: clinical/src/test/kotlin/com/hartwig/actin/clinical/feed/ClinicalFeedValidationTest.kt
package com.hartwig.actin.clinical.feed

import com.hartwig.actin.clinical.curation.TestAtcFactory
import com.hartwig.actin.clinical.feed.ClinicalFeedValidation.validate
import com.hartwig.actin.clinical.feed.patient.PatientEntry
import org.junit.Test

class ClinicalFeedValidationTest {
    @Test
    fun standardTestFeedIsValid() {
        validate(TestFeedFactory.createTestClinicalFeed(), TestAtcFactory.createProperAtcModel())
    }

    @Test(expected = IllegalStateException::class)
    fun doNotAllowDuplicatePatients() {
        val basisFeed = TestFeedFactory.createTestClinicalFeed()
        val patients: List<PatientEntry> = listOf(basisFeed.patientEntries[0], basisFeed.patientEntries[0])
        validate(basisFeed.copy(patientEntries = patients), TestAtcFactory.createProperAtcModel())
    }
}